### PR TITLE
Add range-bound inference stage runtime

### PR DESCRIPTION
**Pull request by ewindisch**
## Summary

Adds the production runtime consumer of the subset-stage factory:

- loads one exact contiguous decoder range through `ModelFactory::create_stage`;
- enforces first/middle/last stage input and output roles;
- runs only `forward_layers` for the owned global range, retaining stage-local cache state and returning only activation or final logits.

This is deliberately stacked on draft PR #1451. It provides the local execution boundary; authenticated lifecycle RPC and encoded inter-host transport remain separate follow-on work.

## Validation

- `cargo test -p hyprstream inference_stage::tests --lib -- --nocapture` — 2 passed
- `cargo clippy --workspace --all-targets --release -- -D warnings` via BuildQ — passed

No hardware validation is claimed.

### CODE REVIEW

**Comment by ewindisch** [comment]
Addressed the independent review findings in exact head `1d201ddc6c2e8925a200789362522c6ffe0a8285`:

- `InferenceStage`, `StageTensor`, and `StageOutput` now carry a private `!Send` marker.
- A loaded stage now admits one `StageSequence` / cache timeline, checks monotonic `start_pos`, and rejects a second timeline.
- Added mock-backed execution coverage for endpoint input roles, hidden-vs-logit output, exact global range forwarding, rejected input non-mutation, and sequence ownership.

Validation:
- focused `inference_stage::tests`: 4 passed
- full release Clippy through BuildQ: passed

The draft remains unmerged. This establishes the local runtime boundary only; it does not claim lifecycle RPC, transport, or hardware acceptance. Requesting a fresh independent review of this exact head.

**Comment by ewindisch** [comment]
Independent exact-head re-review has passed at `1d201ddc6c2e8925a200789362522c6ffe0a8285`. The reviewer confirmed the `!Send` stage/output boundary, one checked cache timeline, and mock-backed execution coverage; focused tests passed 4/4.

This is review readiness only. The draft remains unmerged while the remaining service-lifecycle invocation is completed and independently reviewed; no lifecycle RPC, transport, or hardware claim is made here.

**Comment by ewindisch** [comment]
Closing as stale stacked work. Its base is an old stage-loader branch and the underlying stage factory already landed; any remaining lifecycle work must be reintroduced from current main in a deliberate PR.
